src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -451,6 +453,32 @@ impl Coordinator {
                 &compute_instance.to_string(),
             ])
             .inc();
+        if isolation_level == &IsolationLevel::StrictSerializable && real_time_recency_ts.is_none()
+        {
+            let serializable_det = self.determine_timestamp_for(
+                self.catalog().state(),
+                session,
+                id_bundle,
+                when,
+                compute_instance,
+                timeline_context,
+                real_time_recency_ts,
+                &IsolationLevel::Serializable,
+            )?;
+            match (
+                det.timestamp_context.timestamp(),
+                serializable_det.timestamp_context.timestamp(),
+            ) {
+                (Some(strict), Some(serializable)) => self
+                    .metrics
+                    .timestamp_difference_for_strict_serializable
+                    .with_label_values(&[&compute_instance.to_string()])
+                    .observe(
+                        f64::cast_lossy(u64::from(strict.saturating_sub(*serializable))) / 1000.0,
```

Comment:
Why do we divide by `1000` here? To convert from ms to s? I think it might be more useful to have ms as a metric.

Reply:
Yea, to convert from ms to s. I agree ms is more natural; I chose seconds b/c the other metrics defined in the adapter all use seconds.

Reply:
I concur that for this case ms is desired, because we expect this often be < 1000ms.

Reply:
I put up a commit for this (https://github.com/MaterializeInc/materialize/pull/20528/commits/b5d70fc902d2ce7c377877a71bf522be2d7bccbd) but it doesn't change the granularity of what we're storing. I would keep it in seconds: in my experience mixing metrics of different units leads to a little bit of sadness esp since the current metrics don't seem to include their units

Reply:
> doesn't change the granularity of what we're storing

I don't see where this is in the code. The current diff subtracts two u64s, converts it to a f64, and stores that, so it does seem to be storing this as milliseconds?

Reply:
My point about the granularity is that we're either storing seconds as a float or milliseconds as a float but these are effectively equivalent: we don't store the milliseconds at a finer granularity before or after